proposed/2021/VisualStudioPackageInstallDeepLink.md

```diff
@@ -0,0 +1,163 @@
+# Visual Studio Package Install Deep Link
+
+- Author Name: Chris Gill ([chgill-MSFT](https://github.com/chgill-MSFT))
+- Start Date: 07/19/2021
+- GitHub Issue: https://github.com/NuGet/Home/issues/11032
+- GitHub PR: N/A
+
+## Summary
+
+Support deep links for NuGet.org packages that, when clicked, open the package in the Visual Studio NuGet Package Manager UI (PMUI) for a 1 click install experience.
+
+Applications:
+
+* Add “Install with Visual Studio” link to package README, documentation, or tutorials
+* Deep link suggested alternative package for deprecated packages in Visual Studio for an easier transition experience
+* Share install link with coworkers
+* Include “Install with Visual Studio” option on NuGet.org
+
+## Motivation 
+
+Today, customers must copy a CLI command to install a NuGet package to their project from NuGet.org, or search for the package in Visual Studio. We should enable customers to directly install the package from the Visual Studio package manager UI so they can continue to use their preferred package management tool.
+
+From our NuGet.org survey, 70% of our respondents reported using the Visual Studio NuGet UI to install packages. Additionally, 87% of our respondents use Visual Studio for developing .NET apps. 
+
+Since users prefer installing packages via the VS package manager UI by a large margin relative to CLI methods, we should support that as a first-class experience.
+
+## Success metric
+
+We will use telemetry the measure % of VS package installs that come from a deep link and surveys to understand satisfaction with available installation options for NuGet packages.
+
+## Scope
+
+* Generate a deep link URL that opens the NuGet package in the Visual Studio PMUI
+* Handler is just for Visual Studio
+* Only NuGet.org packages are supported
+
+## Scenarios
```

Comment:
0. Visual Studio isn't installed
I recommend spec'ing this out to be clear.  I would assume (see later comments of mine) that you'd probably NOT show the option if VS isn't installed, but call this scenario out.

Reply:
Out of curiosity, how would a website (such as nuget.org, azure devops, artifactory, etc) detect whether or not VS is installed?  Does the browser have APIs to determine whether or not a specific protocol handler is installed?

Reply:
It looks `github.com` has that integration already. If you click on `Open with Visual Studio` on `NuGet.Client` repo for example it opens a `Microsoft Visual Studio Web Protocol Handler` and user must select the VS version, then the repo is cloned locally.

![image](https://user-images.githubusercontent.com/52756182/132895174-f2f367b0-5584-4b66-b345-258af9fcf2ce.png)

Reply:
The "choose which version of VS" looks like the standard Windows prompt, not github.  If you have multiple monitors, move your browser to the non-primary monitor, and see that the dialog appears on the primary screen.

I don't have a WIndows machine available right now to validate that the "Open with Visual Studio" is not displayed/hidden when VS is not installed, which is what the suggestion is.